src/instruments.rs

```diff
@@ -130,10 +130,27 @@ fn get_macos_version() -> Result<Version> {
         return Err(anyhow!("macOS version cannot be determined"));
     }
 
-    let version_string = std::str::from_utf8(&stdout)?;
-    Version::parse(&version_string).map_err(|error| {
-        anyhow!("cannot parse version: `{}`, because of {}", version_string, error)
-    })
+    semver_from_utf8(&stdout)
+}
+
+/// Returns a semver given a slice of bytes
+///
+/// This function tries to construct a semver struct given a raw utf8 byte array
+/// that may not contain a patch number, `"11.1"` is parsed as `"11.1.0"`.
+fn semver_from_utf8(version: &[u8]) -> Result<Version> {
+    let to_semver = |version_string: &str| {
+        Version::parse(&version_string).map_err(|error| {
+            anyhow!("cannot parse version: `{}`, because of {}", version_string, error)
+        })
+    };
+
+    let version_string = std::str::from_utf8(version)?;
+    match version_string.split('.').count() {
+        1 => to_semver(&format!("{}.0.0", version_string.trim_end_matches('\n'))),
+        2 => to_semver(&format!("{}.0", version_string.trim_end_matches('\n'))),
```

Comment:
I'd probably prefer just `trim()` here, not going to block on that though. :)

Reply:
Short and sweet!